src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -122,66 +160,146 @@ trait GraphDAO {
+      case (key, None) =>
+        // no value for the key so remove anything that is there
+        // can't tell if this is a value or a map anymore so just remove both (there should only be one)
+        vertex.removeProperty(key)
+        vertex.getVertices(Direction.OUT, key).foreach(removeMapVertex)
       case (key, Some(value: DateTime)) => vertex.setProperty(key, value.toDate)
+      case (key, Some(value: Map[_,_])) => serializeAttributeMap(vertex, key, value.asInstanceOf[Map[String, Attribute]], graph, workspaceName)
+      case (key, Some(value: AttributeEntityReference)) => serializeReference(vertex, key, value, graph, workspaceName)
+      case (key, Some(seq: Seq[_])) => seq.headOption match {
+        case None => // empty, do nothing
+        case Some(x: Workflow) =>
+          serializeDomainObjects(vertex, seq.asInstanceOf[Seq[Workflow]], workflowEdge, (w: Workflow) => w.workflowId, graph, workspaceName)
+          val workspaceVertex = getSinglePipelineResult(workspacePipeline(graph, workspaceName.namespace, workspaceName.name)).get
+          new GremlinPipeline(vertex).out(workflowEdge).linkIn(workflowEdge, workspaceVertex).iterate()
+        case Some(x: WorkflowFailure) => serializeDomainObjects(vertex, seq.asInstanceOf[Seq[WorkflowFailure]], workflowFailureEdge, (wf: WorkflowFailure) => (wf.entityType, wf.entityName), graph, workspaceName)
+        case x => throw new RawlsException(s"Unexpected object of type [${x.getClass}] in Seq for attribute [${key}]: [${x}]")
+      }
+      case (key, Some(mrConfig: MethodRepoConfiguration)) => serializeDomainObjects(vertex, Seq(mrConfig), methodRepoConfigEdge, thereShouldOnlyBeOne, graph, workspaceName)
+      case (key, Some(mrMethod: MethodRepoMethod)) => serializeDomainObjects(vertex, Seq(mrMethod), methodRepMethodEdge, thereShouldOnlyBeOne, graph, workspaceName)
+      case (key, Some(ws: WorkspaceName)) => // don't serialize workspace name objects
+
+      // catchall, treat it as something that can be directly set on the vertex, this works for AnyVal and String
+      // but other things may cause orientdb exceptions
       case (key, Some(value)) => vertex.setProperty(key, value)
     })
 
-    vertex.setProperty("_clazz", obj.getClass.getSimpleName)
     vertex
   }
 
   /**
-   * Convert a graph vertex to an instance of T
-   * @param vertex
-   * @param otherProperties properties required to construct T but are not in the vertex. Keys must match the
-   *                        names of the fields on T to set
-   * @tparam T
-   * @return an instance of T populated from the vertex and otherProperties
+   * special function passed into idFxn of serializeDomainObjects when there should only be
+   * one object and thus any existing should be overwritten
    */
-  def fromVertex[T: TypeTag](vertex: Vertex, otherProperties: Map[String, Any]): T = {
-    fromPropertyMap(otherProperties ++ getVertexProperties(vertex))
-  }
+  private def thereShouldOnlyBeOne(x: Any) = 1
```

Comment:
Missed opportunity for a Highlander reference.

![highlander-victory](https://cloud.githubusercontent.com/assets/2701406/9071181/e623fc34-3ac1-11e5-82ca-710808b75225.jpg)